tasks/root.yml

```diff
@@ -4,13 +4,13 @@
   ansible.builtin.user:
     name: root
     password: "{{ users_root_password | ansible.builtin.password_hash('sha512', users_root_password_salt) }}"
-  when: not users_root_password_is_hashed
+  when: not users_root_password_is_hashed and users_root_password is defined and users_root_password_salt is defined
```

Comment:
I would remove the ` and users_root_password_salt is defined` part here.

If someone defines `users_root_password`, I assume they want this role to set the root password, not just silently skip it because they forgot to set `users_root_password_salt`.

For such cases, it's IMO better to explicitly fail and have the role user immediately know that they either need to also set `users_root_password`, or set `users_root_password_is_hashed` to `true`.

Reply:
Thanks!